src/util/detect-bugs/transitionrun-loop.js

```diff
@@ -1,21 +1,26 @@
 // In Safari < 18.2, transitioning custom properties of syntax `*` or `<string>`
 // causes an infinite loop of `transitionrun` or `transitionstart` events.
 // We use this test to detect the bug.
-let dummy = document.createElement("div");
-document.body.appendChild(dummy);
-let property = "--bar-" + Date.now();
-dummy.style.cssText = `${property}: 1; transition: ${property} 1ms step-start allow-discrete`;
+let document = globalThis.document;
+let dummy;
+if (document) {
+	dummy = document.createElement("div");
+	document.body.appendChild(dummy);
+	let property = "--bar-" + Date.now();
+	dummy.style.cssText = `${property}: 1; transition: ${property} 1ms step-start allow-discrete`;
+}
 
 /**
  * Detect if the browser is affected by the Safari transition loop bug.
  * @see https://bugs.webkit.org/show_bug.cgi?id=279012
  * @type {Promise<boolean>}
  */
 export default new Promise(resolve => {
+	if (!document) return resolve(false);
 	let eventsCount = 0;
-	requestAnimationFrame(() => {
+	globalThis.requestAnimationFrame(() => {
```

Comment:
Is this still needed since we exit early if there's no DOM?

Reply:
Yeah no, removing!

Reply:
Done in https://github.com/LeaVerou/style-observer/pull/109/commits/b2370446887298468e041935fbd1d6b5c6c50ff4